### PR TITLE
[SPARK-48296][SQL] Codegen Support for `to_xml`

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The PR adds `Codegen Support` for `to_xml`.

### Why are the changes needed?
Improve codegen coverage.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
- Add new UT & existed UT.
- Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/XmlFunctionsSuite.scala [hunk 1]
```diff
@@ -55,7 +55,7 @@ class XmlFunctionsSuite extends QueryTest with SharedSparkSession {
     val options = Map("rowTag" -> "foo").asJava
 
     checkAnswer(
-      df.select(from_xml($"value", schema)),
+      df.select(from_xml($"value", schema, options)),
```
Comment (panbingkun) [inline sql/core/src/test/scala/org/apache/spark/sql/XmlFunctionsSuite.scala:58]: Obviously, its usage does not match the `title` of UT

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/xmlExpressions.scala [hunk 2]
```diff
@@ -186,9 +186,6 @@ case class SchemaOfXml(
   @transient
   private lazy val xmlOptions = new XmlOptions(options, "UTC")
 
-  @transient
-  private lazy val xmlFactory = xmlOptions.buildXmlFactory()
```
Comment (panbingkun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/xmlExpressions.scala:190]: This variable `xmlFactory` is not used in the expression `SchemaOfXml`, so it is deleted here. It has no strong relationship with this PR.

**Comment by panbingkun** [comment]
cc @yaooqinn @cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/xmlExpressions.scala [hunk 3]
```diff
@@ -298,40 +294,40 @@ case class StructsToXml(
...
+  }
 
   override def inputTypes: Seq[AbstractDataType] = StructType :: Nil
 
   override def prettyName: String = "to_xml"
 
   override protected def withNewChildInternal(newChild: Expression): StructsToXml =
     copy(child = newChild)
+
+  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
```
Comment (yaooqinn) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/xmlExpressions.scala:316]: Simply
```scala
 override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
    val expr = ctx.addReferenceObj("this", this)
    defineCodeGen(ctx, ev, input => s"(UTF8String) $expr.nullSafeEval($input)")
  }
```

Comment (panbingkun) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/xmlExpressions.scala:315]: yeah, updated.
thanks!

Comment (yaooqinn) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/xmlExpressions.scala:315]: We can also revert other refactoring?

Comment (panbingkun) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/xmlExpressions.scala]: Done.

File: sql/core/src/test/scala/org/apache/spark/sql/XmlFunctionsSuite.scala [hunk 4]
```diff
@@ -383,6 +384,22 @@ class XmlFunctionsSuite extends QueryTest with SharedSparkSession {
     }
   }
 
+  test("to_xml - Codegen Support") {
```
Comment (yaooqinn) [inline sql/core/src/test/scala/org/apache/spark/sql/XmlFunctionsSuite.scala:387]:
```suggestion
  test("SPARK-48296: to_xml - Codegen Support") {
```

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
Thanks @panbingkun, merged to master

**Pull request closed without merging**